[Observability Onboarding] Add AWS Firehose quickstart (#4433)
elastic/observability-docs: docs/en/observability/quickstarts/collect-data-with-aws-firehose.asciidoc

[[collect-data-with-aws-firehose]]
= Quickstart: Collect data with AWS Firehose

preview::[]

In this quickstart guide, you'll learn how to send logs and metrics to Elastic using AWS Firehose.

> The AWS Firehose receiver is based on a CloudFormation template that can collect all available CloudWatch logs and metrics for your AWS account.

Reviewer: > The AWS Firehose receiver is based on a CloudFormation template

I'd suggest changing this to something like "The AWS Firehose streams are created using CloudFormation template ...", this better reflects what's going on

> This approach requires minimal configuration as the CloudFormation template creates a Firehose stream, enables CloudWatch metrics collection, and sets up account level subscription filter for CloudWatch log groups to send logs to Elastic via Firehose.

Reviewer 2: ```suggestion
This approach requires minimal configuration as the CloudFormation template creates a Firehose stream, enables CloudWatch metrics collection from all namespaces, and sets up account level subscription filter for CloudWatch log groups to send logs to Elastic via Firehose.
```

> NOTE: Optionally, you can download the CloudFormation template to review and modify the default settings. For more information, check the 

Reviewer: Here I'd also clarify that downloaded template could be used in customer's existing IaC (infrastructure as code) setup.

> link:https://www.elastic.co/docs/current/integrations/aws[AWS integration] documentation.

Reviewer: I don't think this is correct. AWS integration and the template aren't really connected in this context, I'd suggest to separate them and have AWS Integration reference as a separate paragraph that states that some AWS services need additional manual configuration in order to properly ingest logs/metrics, it's an important point we need to emphasize.

Reviewer 2: Agree, we should have a separate paragraph for all optional parameters and how to use them.

Author: I reworded the text based on your comments. Please review and see if it is any better.

[[collect-data-with-aws-firehose]]
= Quickstart: Collect data with AWS Firehose

preview::[]

> In this quickstart guide, you'll learn how to send logs and metrics to Elastic using AWS Firehose.

Reviewer 3: I think we can be more direct but updating this to:

```suggestion
In this quickstart guide, you'll learn how to use AWS Firehose to send logs and metrics to Elastic.
```

> The AWS Firehose streams are created using the CloudFormation template that can collect all available CloudWatch logs and metrics for your AWS account.

Reviewer 3: ```suggestion
The AWS Firehose streams are created using a CloudFormation template, which can collect all available CloudWatch logs and metrics for your AWS account.
```

> This approach requires minimal configuration as the CloudFormation template creates a Firehose stream, enables CloudWatch metrics collection from all namespaces, and sets up account level subscription filter for CloudWatch log groups to send logs to Elastic via Firehose 
> Optionally, you can run AWS CLI command or upload the template to AWS CloudFormation portal on Amazon to change the values for the parameters.

Reviewer 3: ```suggestion
This approach requires minimal configuration as the CloudFormation template creates a Firehose stream, enables CloudWatch metrics collection across all namespaces, and sets up an account-level subscription filter for CloudWatch log groups to send logs to Elastic via Firehose.
Optionally, you can use an AWS CLI command or upload the template to the AWS CloudFormation portal to customize parameter values.
```

IMPORTANT: Some AWS services need additional manual configuration to properly ingest logs and metrics. For more information, check the 
link:https://www.elastic.co/docs/current/integrations/aws[AWS integration] documentation.

Data collection with AWS Firehose is supported on ESS deployments in AWS, Azure and GCP. 

[discrete]

> == Prerequisites

Reviewer: @alaudazzi could we please also include a note that be default CloudFormation stack will be created in the AWS region selected for the user's account, and this can be changed either in AWS Console UI or by adding a `--region` parameter to the AWS CLI command while creating the stack.

* {kibana-ref}/kibana-privileges.html[**Kibana**]: `{ spaces: ['*'], feature: { fleet: ['all'], fleetv2: ['all'] } }`
====
* An active AWS account and the necessary permissions to create delivery streams.

[discrete]
== Limitations

The AWS Firehose receiver has the following limitations:

* It does not support AWS PrivateLink.
* It is not available for on-premise Elastic Stack deployments. 

[discrete]
== Collect your data

. In {kib}, go to **Observability** and click **Add Data**.

. Select **Cloud**, **AWS**, and then select **AWS Firehose**.
+
[role="screenshot"]
image::images/quickstart-aws-firehose-entry-point.png[AWS Firehose entry point]

. Click **Create Firehose Stream in AWS** to create a CloudFormation stack from the CloudFormation template.

. Go back to the **Add Observability Data** page.

[discrete]
== Visualize your data

After installation is complete and all relevant data is flowing into Elastic,
the **Visualize your data** section allows you to access the different dashboards for the various services.

[role="screenshot"]

> image::images/quickstart-aws-firehose-dashboards.png[AWS Firehose dashboards]

Reviewer 3: We kind of just drop the user into the abyss here. Instead of adding a screenshot of the various dashboards, would it be better to describe them and why they might be useful?

Reviewer 3: Maybe we don't have to describe all of them. But I think it'd be helpful to provide an example or two of what dashboards are available and why a user might want to look at them.

Reviewer: Yeah, I still think there is no need to give an exhaustive mapping between each AWS service and a dashboard we provide, but having a single example of could be useful in case it's a new user and they have not used our dashboards before.

Here is an screenshot for the "VPC Flow" logs dashboard
![CleanShot 2024-11-04 at 10 19 23@2x](https://github.com/user-attachments/assets/a8ef63bf-d523-4480-bdfb-2cb2bbe53ddf)

Reviewer 3: Sounds good to me 👍

> Refer to <<observability-introduction>> for a description of other useful features.

Reviewer 3: I'm not sure how this is useful.

Author: I guess it's exactly the opposite, from the specific QS, the link takes you to a more general overview of Elastic Observability, which I believe is useful here. Also, this link appears on the other quickstarts ([Quickstart: Monitor hosts with Elastic Agent](https://www.elastic.co/guide/en/observability/8.16/quickstart-monitor-hosts-with-elastic-agent.html) and [Quickstart: Monitor your Kubernetes cluster with Elastic Agent](https://www.elastic.co/guide/en/observability/8.16/monitor-k8s-logs-metrics-with-elastic-agent.html).
